diff -urN dnsmasq-2.30/dbus/dnsmasq.conf dnsmasq-2.30-patched/dbus/dnsmasq.conf
--- dnsmasq-2.30/dbus/dnsmasq.conf	1969-12-31 18:00:00.000000000 -0600
+++ dnsmasq-2.30-patched/dbus/dnsmasq.conf	2006-05-02 09:52:56.000000000 -0500
@@ -0,0 +1,16 @@
+<!DOCTYPE busconfig PUBLIC
+ "-//freedesktop//DTD D-BUS Bus Configuration 1.0//EN"
+ "http://www.freedesktop.org/standards/dbus/1.0/busconfig.dtd">
+<busconfig>
+	<policy user="root">
+		<allow own="uk.org.thekelleys.dnsmasq"/>
+		<allow send_destination="uk.org.thekelleys.dnsmasq"/>
+                <allow send_interface="uk.org.thekelleys.dnsmasq"/>
+	</policy>
+	<policy context="default">
+                <deny own="uk.org.thekelleys.dnsmasq"/>
+                <deny send_destination="uk.org.thekelleys.dnsmasq"/>
+                <deny send_interface="uk.org.thekelleys.dnsmasq"/>
+        </policy>
+</busconfig>
+
