diff -urN dnsmasq-2.43/contrib/dnslist/dnslist.pl dnsmasq-2.43-patched/contrib/dnslist/dnslist.pl
--- dnsmasq-2.43/contrib/dnslist/dnslist.pl	2008-06-02 13:58:52.000000000 -0500
+++ dnsmasq-2.43-patched/contrib/dnslist/dnslist.pl	2008-07-16 14:23:16.000000000 -0500
@@ -44,7 +44,7 @@
 
 # Location of state file.  (This is the dnsmasq default.)
 # Change with -s <file>
-my $dnsmasq_state_file = '/var/lib/misc/dnsmasq.leases';
+my $dnsmasq_state_file = '/var/lib/dnsmasq/dnsmasq.leases';
 # Location of template.  (Assumed to be in current directory.)
 # Change with -t <file>
 my $html_template_file = 'dnslist.tt2';
diff -urN dnsmasq-2.43/dnsmasq.conf.example dnsmasq-2.43-patched/dnsmasq.conf.example
--- dnsmasq-2.43/dnsmasq.conf.example	2008-06-02 13:58:52.000000000 -0500
+++ dnsmasq-2.43-patched/dnsmasq.conf.example	2008-07-16 14:23:29.000000000 -0500
@@ -351,7 +351,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+#dhcp-leasefile=/var/lib/dnsmasq/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
diff -urN dnsmasq-2.43/man/dnsmasq.8 dnsmasq-2.43-patched/man/dnsmasq.8
--- dnsmasq-2.43/man/dnsmasq.8	2008-07-10 10:48:14.000000000 -0500
+++ dnsmasq-2.43-patched/man/dnsmasq.8	2008-07-16 14:22:35.000000000 -0500
@@ -1116,7 +1116,7 @@
 
 .IR /etc/ethers
 
-.IR /var/lib/misc/dnsmasq.leases 
+.IR /var/lib/dnsmasq/dnsmasq.leases 
 
 .IR /var/db/dnsmasq.leases
 
diff -urN dnsmasq-2.43/man/es/dnsmasq.8 dnsmasq-2.43-patched/man/es/dnsmasq.8
--- dnsmasq-2.43/man/es/dnsmasq.8	2008-06-02 13:58:52.000000000 -0500
+++ dnsmasq-2.43-patched/man/es/dnsmasq.8	2008-07-16 14:22:48.000000000 -0500
@@ -1145,7 +1145,7 @@
 
 .IR /etc/ethers
 
-.IR /var/lib/misc/dnsmasq.leases 
+.IR /var/lib/dnsmasq/dnsmasq.leases 
 
 .IR /var/db/dnsmasq.leases
 
diff -urN dnsmasq-2.43/man/fr/dnsmasq.8 dnsmasq-2.43-patched/man/fr/dnsmasq.8
--- dnsmasq-2.43/man/fr/dnsmasq.8	2008-07-11 03:28:28.000000000 -0500
+++ dnsmasq-2.43-patched/man/fr/dnsmasq.8	2008-07-16 14:22:54.000000000 -0500
@@ -1280,7 +1280,7 @@
 
 .IR /etc/ethers
 
-.IR /var/lib/misc/dnsmasq.leases 
+.IR /var/lib/dnsmasq/dnsmasq.leases 
 
 .IR /var/db/dnsmasq.leases
 
diff -urN dnsmasq-2.43/src/config.h dnsmasq-2.43-patched/src/config.h
--- dnsmasq-2.43/src/config.h	2008-07-11 03:44:17.000000000 -0500
+++ dnsmasq-2.43-patched/src/config.h	2008-07-16 14:21:02.000000000 -0500
@@ -43,7 +43,7 @@
 #elif defined(__sun__) || defined (__sun)
 #   define LEASEFILE "/var/cache/dnsmasq.leases"
 #else
-#   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#   define LEASEFILE "/var/lib/dnsmasq/dnsmasq.leases"
 #endif
 #if defined(__FreeBSD__)
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
@@ -167,7 +167,7 @@
 #define HAVE_TFTP
 /* #define HAVE_BROKEN_RTC */
 /* #define HAVE_ISC_READER */
-/* #define HAVE_DBUS */
+#define HAVE_DBUS
 
 #if defined(HAVE_BROKEN_RTC) && defined(HAVE_ISC_READER)
 #  error HAVE_ISC_READER is not compatible with HAVE_BROKEN_RTC
