From cfcad42ff1ddee8e64d120f18016a654152d0215 Mon Sep 17 00:00:00 2001
From: Simon Kelley <simon@thekelleys.org.uk>
Date: Fri, 17 May 2013 11:32:03 +0100
Subject: [PATCH] Fix failure to start with ENOTSOCK

---
 CHANGELOG     | 6 ++++++
 src/dnsmasq.c | 2 +-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/CHANGELOG b/CHANGELOG
index 7aa0024..48b6070 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -31,7 +31,13 @@ version 2.67
 	    want to continue to bind the aliases too, you need to add
 	    eg. --interface=eth0:0 to the config. 
 	
+	    Fix "failed to set SO_BINDTODEVICE on DHCP socket: Socket 
+	    operation on non-socket" error on startup with
+	    configurations which have exactly one --interface option
+	    and do RA but _not_ DHCPv6. Thanks to Trever Adams for the
+	    bug report.
 
+		
 version 2.66
             Add the ability to act as an authoritative DNS
             server. Dnsmasq can now answer queries from the wider 'net
diff --git a/src/dnsmasq.c b/src/dnsmasq.c
index 43b8cb1..b0f984d 100644
--- a/src/dnsmasq.c
+++ b/src/dnsmasq.c
@@ -248,7 +248,7 @@ int main (int argc, char **argv)
 #endif
 
 #if defined(HAVE_LINUX_NETWORK) && defined(HAVE_DHCP6)
-      if (daemon->dhcp6)
+      if (daemon->doing_dhcp6)
 	bindtodevice(daemon->dhcp6fd);
 #endif
     }
-- 
1.8.1.4

