diff -up dnsmasq-2.41test20/src/config.h.orig dnsmasq-2.41test20/src/config.h
--- dnsmasq-2.41test20/src/config.h.orig	2007-11-27 14:09:01.000000000 -0600
+++ dnsmasq-2.41test20/src/config.h	2007-12-04 08:21:49.000000000 -0600
@@ -42,7 +42,7 @@
 #elif defined(__sun__)
 #   define LEASEFILE "/var/cache/dnsmasq.leases"
 #else
-#   define LEASEFILE "/var/lib/misc/dnsmasq.leases"
+#   define LEASEFILE "/var/lib/dnsmasq/dnsmasq.leases"
 #endif
 #if defined(__FreeBSD__)
 #   define CONFFILE "/usr/local/etc/dnsmasq.conf"
@@ -182,7 +182,7 @@ NOTES:
 #define HAVE_TFTP
 /* #define HAVE_BROKEN_RTC */
 /* #define HAVE_ISC_READER */
-/* #define HAVE_DBUS */
+#define HAVE_DBUS
 
 #if defined(HAVE_BROKEN_RTC) && defined(HAVE_ISC_READER)
 #  error HAVE_ISC_READER is not compatible with HAVE_BROKEN_RTC
diff -up dnsmasq-2.41test20/man/es/dnsmasq.8.orig dnsmasq-2.41test20/man/es/dnsmasq.8
--- dnsmasq-2.41test20/man/es/dnsmasq.8.orig	2007-11-27 14:09:01.000000000 -0600
+++ dnsmasq-2.41test20/man/es/dnsmasq.8	2007-12-04 08:33:07.000000000 -0600
@@ -1075,7 +1075,7 @@ un CPU de 1GHz y aproximadamente 60MB de
 
 .IR /etc/ethers
 
-.IR /var/lib/misc/dnsmasq.leases 
+.IR /var/lib/dnsmasq/dnsmasq.leases 
 
 .IR /var/db/dnsmasq.leases
 
diff -up dnsmasq-2.41test20/man/dnsmasq.8.orig dnsmasq-2.41test20/man/dnsmasq.8
--- dnsmasq-2.41test20/man/dnsmasq.8.orig	2007-11-27 14:09:01.000000000 -0600
+++ dnsmasq-2.41test20/man/dnsmasq.8	2007-12-04 08:33:07.000000000 -0600
@@ -1070,7 +1070,7 @@ file needs a 1GHz processor and about 60
 
 .IR /etc/ethers
 
-.IR /var/lib/misc/dnsmasq.leases 
+.IR /var/lib/dnsmasq/dnsmasq.leases 
 
 .IR /var/db/dnsmasq.leases
 
diff -up dnsmasq-2.41test20/dnsmasq.conf.example.orig dnsmasq-2.41test20/dnsmasq.conf.example
--- dnsmasq-2.41test20/dnsmasq.conf.example.orig	2007-11-27 14:09:01.000000000 -0600
+++ dnsmasq-2.41test20/dnsmasq.conf.example	2007-12-04 08:44:16.000000000 -0600
@@ -344,7 +344,7 @@
 # The DHCP server needs somewhere on disk to keep its lease database.
 # This defaults to a sane location, but if you want to change it, use
 # the line below.
-#dhcp-leasefile=/var/lib/misc/dnsmasq.leases
+#dhcp-leasefile=/var/lib/dnsmasq/dnsmasq.leases
 
 # Set the DHCP server to authoritative mode. In this mode it will barge in
 # and take over the lease for any client which broadcasts on the network,
@@ -461,4 +461,4 @@
 
 # Include a another lot of configuration options.
 #conf-file=/etc/dnsmasq.more.conf
-#conf-dir=/etc/dnsmasq.d
+conf-dir=/etc/dnsmasq.d
