diff -up dnsmasq-2.72/dnsmasq.conf.example.syntax_err dnsmasq-2.72/dnsmasq.conf.example
--- dnsmasq-2.72/dnsmasq.conf.example.syntax_err	2014-10-06 10:30:31.841567305 +0200
+++ dnsmasq-2.72/dnsmasq.conf.example	2014-10-06 10:33:51.195697413 +0200
@@ -645,4 +645,4 @@
 #conf-dir=/etc/dnsmasq.d,.bak
 
 # Include all files in a directory which end in .conf
-#conf-dir=/etc/dnsmasq.d/*.conf
\ No newline at end of file
+#conf-dir=/etc/dnsmasq.d/,*.conf
