diff -urN dnsmasq-2.33/src/config.h dnsmasq-2.33-patched/src/config.h
--- dnsmasq-2.33/src/config.h	2006-08-04 06:02:36.000000000 -0500
+++ dnsmasq-2.33-patched/src/config.h	2006-08-15 15:14:19.000000000 -0500
@@ -156,7 +156,7 @@
 /* platform independent options- uncomment to enable */
 /* #define HAVE_BROKEN_RTC */
 /* #define HAVE_ISC_READER */
-/* #define HAVE_DBUS */
+#define HAVE_DBUS
 
 #if defined(HAVE_BROKEN_RTC) && defined(HAVE_ISC_READER)
 #  error HAVE_ISC_READER is not compatible with HAVE_BROKEN_RTC
